src/test/fuzz/addrman.cpp

```diff
@@ -66,23 +66,7 @@ CNetAddr RandAddr(FuzzedDataProvider& fuzzed_data_provider, FastRandomContext& f
     if (fuzzed_data_provider.remaining_bytes() > 1 && fuzzed_data_provider.ConsumeBool()) {
         addr = ConsumeNetAddr(fuzzed_data_provider);
     } else {
-        // The networks [1..6] correspond to CNetAddr::BIP155Network (private).
-        static const std::map<uint8_t, uint8_t> net_len_map = {{1, ADDR_IPV4_SIZE},
-                                                               {2, ADDR_IPV6_SIZE},
-                                                               {4, ADDR_TORV3_SIZE},
-                                                               {5, ADDR_I2P_SIZE},
-                                                               {6, ADDR_CJDNS_SIZE}};
-        uint8_t net = fast_random_context.randrange(5) + 1; // [1..5]
-        if (net == 3) {
-            net = 6;
-        }
-
-        CDataStream s(SER_NETWORK, PROTOCOL_VERSION | ADDRV2_FORMAT);
-
-        s << net;
-        s << fast_random_context.randbytes(net_len_map.at(net));
-
-        s >> addr;
+        addr = ConsumeNetAddr(fuzzed_data_provider, &fast_random_context);
```

Comment:
I don't think consuming addresses from `FastRandomContext` has much upside in fuzzing. Especially after this PR since `ConsumeNetAddr` will now produce addrs of any kind. I think you were the author of `RandAddr` so maybe you can elaborate on why you added this back then.

Reply:
Yes, because the addrman tests (that use `RandAddr()`) need many many many addresses. If all are generated from the fuzzer input, they exhaust it very quickly.

Reply:
Here is the relevant discussion wrt generating addresses from `FastRandomContext`: https://github.com/bitcoin/bitcoin/pull/21129#discussion_r675972823

Reply:
Added a comment to `ConsumeNetAddr()` to explain this.

Reply:
Ah I see, thanks for linking!

I'm still skeptical that consuming from `FastRandomContext` is useful in this context and I would speculate that leaving the addrs blank or providing some valid dummies is probably just as useful.

In any case, I guess this isn't really the concern of this PR. I was about to suggest that you add a comment to `ConsumeNetAddr` stating that for most targets not consuming from `FastRandomContext` is preferable but I see that you just did that on the latest push :)

Reply:
The test that fills addrman needs different addresses. This is why leaving them blank (e.g. `0.0.0.0`) does not suffice. `FastRandomContext` is a convenient way to generate "unlimited" amount of (mostly) valid, different, dummies ;-)

Reply:
To add to this, the idea of `addrman_serdeser` is to be able to fill up a large fraction of AddrMan, for which we need a large variety of addresses because similar addresses will collide with each other.

The "(mostly) valid," is changed by this a little bit though: I think we'll now fill addrman with less addresses because 1 in 6 addresses will be of type `NET_INTERNAL` which is invalid and not accepted to AddrMan.

Reply:
`RandAddr()` could have returned `NET_INTERNAL` also before (if it called `ConsumeNetAddr()`), but you are right that this has little value because near the end `RandAddr()` will convert any invalid to always-the-same-`5.5.5.5`. I added a few retries to try to avoid that.

Reply:
oh, right, i missed that. Retrying a few times makes sense to me.